docs/tutorials/services/constrain-motion.md

```diff
@@ -0,0 +1,432 @@
+
+Since this tutorial gets a bit more complicated than the last, let's configure a representation of the table so you can see it in the Frame System visualizer.
+This configured table won't be taken into account by the Motion Service, but it's useful to be able to see it.
+
+On your robot's **Config** tab, create a new component called `table` with **Type** `generic` and **Model** `fake`.
+Click **Create component**, then click **Add frame**.
+
+Go to the **Frame System** subtab.
+Click **table** on the left menu.
+For **Geometry** click **box**.
+
+Set the dimensions (**Size**) to `1000` x `1000` x `20`.
+Notice that the top of the table is now 10mm above the origin, because the 20mm thickness is centered on the centerpoint.
+Account for this by giving the table a **Translation** of `-10` in the Z direction, moving the top of the table to 0.
+
+![The Frame System subtab of the Components tab with a 1000 wide, 1000 deep, 200 millimeter thick box representing the top of the table.](../../img/constrain-motion/frame-table.jpg)
+
+### Visualize obstacles on the table
+-->
+
+## Modify your robot's working environment
+
+In the last tutorial, you [defined a table obstacle](../plan-motion-with-arm-gripper/#describe-the-robots-working-environment) to prevent the arm and gripper from hitting the table upon which the arm is mounted.
+In this tutorial, you'll expand on the code that describes your robot's working environment in two ways:
+
+- Define an additional obstacle: a tissue box sitting on the table.
+- Add a `z_offset` parameter to the `table_origin` and to the new `box_origin`.
+  This makes it easier to calibrate your motion plans based on how high or low your arm is mounted compared to the table.
+
+The following example shows the two obstacles defined with the expanded configuration.
+Adjust the dimensions and positions of the obstacles to describe your own scenario:
+
+```python {class="line-numbers linkable-line-numbers"}
+# Use this offset to set your z to calibrate based on where your table actually is
+z_offset = 10
+
+# Create a table obstacle
+table_origin = Pose(x=0.0, y=0.0, z=-19.0+z_offset)
+table_dims = Vector3(x=2000.0, y=2000.0, z=38.0)
+table_object = Geometry(center=table_origin, box=RectangularPrism(dims_mm=table_dims))
+
+# Create a tissue box obstacle
+box_origin = Pose(x=400, y=0, z=50+z_offset)
+box_dims = Vector3(x=120.0, y=80.0, z=100.0)
+box_object = Geometry(center=box_origin, box=RectangularPrism(dims_mm=box_dims))
+
+obstacles_in_frame = GeometriesInFrame(reference_frame="world",
+    geometries=[table_object, box_object])
+```
+
+## Determine the gripper's axes
+
+You need to figure out how the gripper's frame corresponds to its hardware.
+That is, you need to determine which axis points from the "wrist" to the end of the gripper, which axis lines up with the direction the jaws actuate, and, when the gripper is holding a cup, which axis passes vertically through the cup.
```

Comment:
Well, in the next paragraph I used the language "points" because there it's talking about the positive Z axis specifically. And used the same language here for consistency. Do you think that's okay? Whereas, it's near impossible to explain succinctly which way the other positive axes are "pointing," so I didn't bother.

Reply:
Agreed that this is definitely a tricky concept to convey. `points` is fine, especially if used consistently. Feel free to keep as is here, it is definitely understandable in present form.